data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorService.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor;
+
+import org.opensearch.dataprepper.plugins.processor.databasedownload.DBSourceOptions;
+import java.net.InetAddress;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation class of geoIP-processor plugin service class.
+ * It is responsible for calling of mmdb files download
+ */
+public class GeoIPProcessorService {
+
+    /**
+     * GeoIPProcessorService constructor for initialization of required attributes
+     * @param geoIPProcessorConfig geoIPProcessorConfig
+     * @param tempPath tempPath
+     */
+    public GeoIPProcessorService(GeoIPProcessorConfig geoIPProcessorConfig, String tempPath) {
+       //TODO
+    }
+
+    /**
+     * Calling downlaod method abased on the database path type
+     * @param DBSourceOptions DBSourceOptions
+     */
+    public void downloadThroughURLandS3(DBSourceOptions DBSourceOptions) {
```

Comment:
Is customer allowed to supply their own .mmdb file?

Reply:
Yes. In case of S3option, customer can place in their own .mmdb in S3 bucket, download and use it in the GeoIP plugin. However for URL option, the .mmdb file will be downloaded automatically from MaxMind portal and user can't supply their own .mmdb.